app/javascript/components/tooltips/Student.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { useRequestQuery } from '../../hooks/request-query'
+import { QueryResult } from 'react-query'
+import { fromNow } from '../../utils/time'
+import { Loading } from '../common/Loading'
+
+export function Student({
+  endpoint,
+  styles,
+}: {
+  endpoint: string
+  styles?: React.CSSProperties
+}) {
+  const request = {
+    endpoint: endpoint,
+    options: { staleTime: 1000 * 60 },
```

Comment:
This property ensure that for 1000ms * 60 (hence: 60 seconds) there won't be no new retrieval of data for the request key. As the request key is unique per tooltip (two students have two different URLs), this means that the data is cached per student for 60 seconds.

Reply:
If you need to comment it here, you probably want to comment it _there_. This is good explanation to have _in_ the source code! 

That said, I do _not_ like this local state caching. `react-query` should optimistically provide the data, and `fetch` should cache for us. Not an _extra_ cache layer right here.

Reply:
> `react-query` should optimistically provide the data, and fetch should cache for us. Not an extra cache layer right here.

How would we do that? By sending the appropriate cache headers from the backend?

Reply:
Scratch that. I've looked at the Rails caching options and have found this: https://github.com/exercism/v3-website/pull/223/commits/ef6ae183cf54491fd26742003b6fc1a9bdd4bb03

Seems to work well:

![image](https://user-images.githubusercontent.com/135246/97548814-e7a1d700-19cf-11eb-9190-544b21d6c07b.png)

Reply:
Sorry. I should have elaborated!

Rails caching instructs HTTP Cache. You can check this by doing a hard reload (cleared cache) and inspect the response. You'll see `200 ok`, and as you see in this response, `max-age=60, private`. This means that the next 60 seconds, it won't even _query_ the server.